### ISSUE TITLE
CapeV2 Azure Client ID# and Secret for az.conf

### ISSUE THREAD

**Issue by WVAuditors** [kevoreilly/CAPEv2#1041]
I have successfully setup a Cape sandbox already in a proxmox home lab. I am trying to use the Azure guide according to the documentation. I believe I may be a little confused about the az machinery. Web interface is up and running no issues there. I am at a stopping point on the following in the az.conf. 

# Client ID for Azure
client_id =

# Secret for Azure
secret =

I know that I have to create a application for it to connect thru in Azure. I understand that much but any help would be appreciated or even if this has been tested and known to work. Any past experiences with setting this up would be greatly appreciated. I love this project and have been using it in proxmox environment for about 3 months now to scan URLs and attachment in our environment at the West Virginia State Auditor's office . I understand you guys are probably busy working on this project but if you could just give me a quick hint anything would be a great help. I searched all the documentation but that part is somewhat MIA at the moment. Thank you so much and your help is very appreciated.

**Comment by cccs-kevin** [comment]
Hey @WVAuditors, I'm running the Azure machinery in a production-level environment right now, and am the main developer behind the machinery. Is the question that you have related to how to create the Azure application?

**Comment by WVAuditors** [comment]
Yes that is the question. I am not sure how to create the Azure application to get the client id & secret for the connection to take place when running python3 cuckoo.py. I have a place to input in az.conf of course but I am super confused about the Azure end of it configuring the app. I know you guys prob don't support the Microsoft side of it but I thought it may be worth the time to pick a brain of someone who has actually setup Cape with Azure.

**Comment by cccs-kevin** [comment]
Gotchya, I'll update the CAPE docs to hopefully assist with this. 

That being said, to create the Azure application, you have to do the following:
1. In the Azure portal, head to "App registrations"
2. Click "New registration"
3. Give the app a name and determine what the supported account types are
4. Create the app
5. Head to the app resource in the portal
6. Select "API permissions" in the sidebar
7. Click "Add a permission"
8. Give the app "User.Read" to the Microsoft Graph API
9. On the sidebar, click "Certificates & secrets"
10. Select "+ New client secret"
11. Create a secret
12. Copy the secret details and paste it into the `az.conf`
13. Select "Overview" on the sidebar
14. You can find the `client_id` and `tenant_id` details here. Copy and paste them into the `az.conf`

Let me know if you have any questions!

**Comment by WVAuditors** [comment]
Hey also what version of django runs on your setup if you dont mind me asking. I am getting this. 

Request Method: | GET
-- | --
http://10.14.1.4:8001/submit/
4.0.1
AttributeError
'Config' object has no attribute 'None'
/opt/CAPEv2/web/submission/views.py, line 536, in <listcomp>
/bin/python3
3.8.10

**Comment by cccs-kevin** [comment]
I'm using Django v4.0.4 with Python3.8.10. I don't use the web GUI though so I'm not too familiar with what could be happening in your scenario.

**Comment by WVAuditors** [comment]
Thank you so much. I will give it a go!

**Comment by cccs-kevin** [comment]
@WVAuditors I've got a PR (https://github.com/kevoreilly/CAPEv2/pull/1042) that will hopefully assist with the setup. Keep the questions coming!

**Comment by kevoreilly** [comment]
Thank you

**Comment by WVAuditors** [comment]
I am having success but I am not sure why the machine in the scale set instance is grabbing an up of 10.14.1.5 from VNET2_SUB1. Shouldn't I be getting DHCP from the VNET2_SUB2 subnet (ex. 10.14.2.x) I was able to correct the issue with Django also. 

Instance ID : 1


Status

[Specialized, 2 more](https://portal.azure.com/#blade/Microsoft_Azure_Compute/VirtualMachineInstanceStatusBlade/id/%2Fsubscriptions%2Ffb6e0b04-4f56-47c6-83dc-0114c7a1e667%2FresourceGroups%2FRG3%2Fproviders%2FMicrosoft.Compute%2FvirtualMachineScaleSets%2Fcuckoo1%2FvirtualMachines%2F1)
Location
:
East US
Provisioning state
:
Succeeded
Latest model applied
:
Yes
Computer name
:
---
Fault domain
:
-
SKU
:
Standard_DS3_v2
Public IP address
:
-
Private IP address
:
10.14.1.5
Public IP address (IPv6)
:
-
Private IP address (IPv6)
:
-
Virtual network/subnet
:
VNET2/VNET2_SUB1
Disk
:
cuckoo1_cuckoo1_1_OsDisk_1_ca0d326d805142748bff1160e7cd8c2d

**Comment by cccs-kevin** [comment]
Hey @WVAuditors , a couple of things:
1. Please improve your copy-paste game in GitHub, it's hard for me to understand the details that you have provided :)
2. To clarify, https://github.com/kevoreilly/CAPEv2/blob/c196919f8c1f2162a40c866132f1d8698fda11e0/conf/az.conf#L31 <- this value should be the subnet VNET2_SUB2. I'll update the conf to be more clear.

**Comment by WVAuditors** [comment]
I apologize for that. I will try to be me clear with my copy and paste next time. I will also try not to bother you guys too much I know you are busy.

**Comment by cccs-kevin** [comment]
> I apologize for that. I will try to be more clear with my copy and paste next time. I will also try not to bother you guys too much I know you are busy.

I don't mean to discourage conversation, it's just easier on my end to understand what your question is and what the root of the issue is :)

**Comment by WVAuditors** [comment]
Understood. I will go back and check all my configs to see I may have transposed something.

**Comment by WVAuditors** [comment]
I made the correction needed. Ran analysis on puttygen.exe to test. 

2022-08-05 15:59:54,365 [lib.cuckoo.core.guest] INFO: Task #8: Analysis completed successfully (id=cuckoo1_0, ip=10.14.2.6)

I do have some questions about the agent.py startup from disk/snapshot (Windows 7 Instance). For some reason even if I move the agent.py to the startup folder at C:\ProgramData\Microsoft\Windows\Start Menu\Programs\StartUp the agent does not startup automatically. I reimaged the instance and still no go. I have to manually start the agent.py for the instance running. Also if I want to run this on a dirty line is there any difference on how I would accomplish this with Azure? I appreciate your feedback and time. Sorry I am struggling a little with this but it is a little different on how the machinery works as compared to my proxmox machinery setup. I appreciate your help.

**Comment by cccs-kevin** [comment]
Okay cool. So the way that snapshots work in Azure (if you are unfamiliar) is that they are captures of the disk data only - not the VM state too.  So to have the agent start on reboot/reimage (reimage is essentially a reset of the disk data and then a reboot), you need the agent to be in the startup folder (or in a scheduled task) when you take the snapshot. 

To confirm that the agent starts correctly, on your golden image VM, ensure that you have the agent.py file in the startup folder and then reboot the VM and check if the agent.py has started and is listening on port 8000 (You can confirm this by `netstat -an | grep 8000`). If the agent starts correctly, you can take a snapshot of the golden image VM's disk and then create an Image using this snapshot in the Compute Gallery.

Regarding the dirty line, depending on your network security group rules are implemented, this should be as easy as setting the `route` to `internet` in `routing.conf` or specifying `route` in the task options of a submission.

**Comment by WVAuditors** [comment]
I was able to get the agent.py script to start on the instance. Thank you for your help. I had to delete the vm scale set , set to exclude prior versions of image from latest. Then used my 0.0.6 version on the scale build. Was able to pull a successful analysis on a test file.  Seems to be working great but I still need to work on my golden image installing my applications. This seems to work better than my proxmox machinery build.

**Comment by cccs-kevin** [comment]
🚀

**Comment by cccs-kevin** [comment]
Sounds good @WVAuditors ! @kevoreilly @doomedraven  this issue can be closed. If you have any follow-up questions make sure to reach out! (Also follow the issue template or else the issue will be automatically closed)
